File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero.py
class Solution(object):
    def largestCombination(self, candidates):
        ans, i = 0, 1
        while i <= 10000000:
            curr = 0
            for n in candidates:
                if n & i > 0:
                    curr += 1
            ans = max(ans, curr)
            i = i << 1
        
        return ans

File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        int ans = 0, curr = 0;
        for(int i = 1; i<=10000000; i <<= 1) {
            curr = 0;
            for(int n : candidates) {
                if((n & i) > 0) curr++;
            }
            ans = max(ans, curr);
        }

        return ans;
    }
};